book/visual-effects.md

```diff
@@ -519,11 +551,236 @@ high-density enough to retire these techniques.
 
 [font-hinting]: https://en.wikipedia.org/wiki/Font_hinting
 
+
+
+Browser compositing
+===================
+
+So far, Skia and SDL have just made our browser more complex, but the
+low-level control offered by these libraries is important because it
+allows us to optimize commonly-used operations like scrolling.
+
+So far, any time the user scrolled a web page, we had to clear
```

Comment:
```suggestion
Any time the user scrolled a web page, we had to clear
```

Reply:
I think the "so far" is important because the rest of the sentence is in past tense.

Reply:
Fair. I just didn't like how it repeated the exact same starting words as the previous paragraph. Is there a way to rephrase?

Reply:
Ah! Previous paragraph can lose its "so far" instead.